### PR TITLE
Update to latest Raiden and contracts

### PR DESCRIPTION
**Pull request by karlb**
Closes https://github.com/raiden-network/raiden-services/issues/555.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,7 @@ install:
 
 install-dev:
 	pip install -r requirements-dev.txt
+	pip install -e .
```
Comment (palango) [inline Makefile]: Nice, I wanted to do this for a long time.

File: src/monitoring_service/states.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
 from dataclasses import dataclass, field
 from typing import Iterable, Optional
 
+from eth_typing.evm import HexAddress
```
Comment (palango) [inline src/monitoring_service/states.py]: Mh, a bit ugly to import this as well. Is this what the contracts use?

File: src/monitoring_service/handlers.py [hunk 3]
```diff
@@ -370,6 +370,10 @@ def updated_head_block_event_handler(event: Event, context: Context) -> None:
 
 
 def _is_mr_valid(monitor_request: MonitorRequest, channel: Channel) -> bool:
+    if monitor_request.non_closing_signer != monitor_request.non_closing_participant:
```
Comment (palango) [inline src/monitoring_service/handlers.py]: Do we have a test for this?

File: src/monitoring_service/database.py [hunk 4]
```diff
@@ -84,11 +84,12 @@ def get_monitor_request(
         kwargs = {
             key: val
             for key, val in zip(row.keys(), row)
-            if key not in ("non_closing_signer", "saved_at", "waiting_for_channel")
```
Comment (palango) [inline src/monitoring_service/database.py]: Was that a bug before?

File: src/monitoring_service/states.py (hunk 2 above)
Comment (karlb) [reply src/monitoring_service/states.py]: Yes, since they don't have access to Raiden's types.

File: src/monitoring_service/handlers.py (hunk 3 above)
Comment (karlb) [reply src/monitoring_service/handlers.py]: I'll add one and move this to the request collector.

File: src/monitoring_service/database.py (hunk 4 above)
Comment (karlb) [reply src/monitoring_service/database.py]: No, we generate the `non_closing_signer` out of the signature when constructing a `MonitorRequest` object. That column is only added to the DB to enforce the correct uniqueness of MRs.

The current handling of `MonitorRequest`, `UnsignedMonitorRequest`, 'non_closing_signer' and `non_closing_participant` is not ideal anymore, but I wanted to keep the changes small for now.

Comment (palango) [reply src/monitoring_service/database.py]: Ok, great!

**Pull request merged**